weldx/transformations.py

```diff
@@ -379,45 +518,449 @@ def _calculate_orthogonal_axis(a_0, a_1):
+
+        :param refernce: Coordinate system that provides the reference time.
+        :return: Coordinate system with interpolated data
+        """
+        if not isinstance(refernce, LocalCoordinateSystem):
+            raise Exception("Invalid reference type")
+        if refernce.time is not None:
+            times = refernce.time
+            return self.interp_time(times)
+        else:
+            raise Exception("Reference coordinate system has no time component")
+
+    def invert(self) -> "LocalCoordinateSystem":
+        """
+        Get a local coordinate system defining the parent in the child system.
+
+        Inverse is defined as basis_new=basis.T, origin_new=basis.T*(-origin)
+        :return: Inverted coordinate system.
+        """
+        basis = ut.xr_transpose_matrix_data(self.basis, dim1="c", dim2="v")
+        origin = ut.xr_matmul(
+            self.basis, -self.origin, dims_a=["c", "v"], dims_b=["c"], trans_a=True
+        )
+        return LocalCoordinateSystem(basis, origin)
+
+
+# coordinate system manager class ------------------------------------------------------
+
+
+class CoordinateSystemManager:
+    """Manages multiple coordinate systems and the transformations between them."""
+
+    @dataclass
+    class CoordinateSystemData:
+        coordinate_system_name: Hashable
+        data: xr.DataArray
+
+    def __init__(self, root_coordinate_system_name: Hashable):
+        """
+        Construct a coordinate system manager.
+
+        :param root_coordinate_system_name: Name of the root coordinate system. This can
+        be any hashable type, but it is recommended to use strings.
+        """
+        self._graph = nx.DiGraph()
+        self._data = dict()
+        self._add_coordinate_system_node(root_coordinate_system_name)
+
+    def _add_coordinate_system_node(self, coordinate_system_name):
+        self._check_new_coordinate_system_name(coordinate_system_name)
+        self._graph.add_node(coordinate_system_name, data=[])
+
+    def _add_edges(
+        self, node_from: Hashable, node_to: Hashable, lcs: LocalCoordinateSystem
+    ):
+        """
+        Add an edge to the internal graph.
+
+        :param node_from: Start node of the edge
+        :param node_to: End node of the edge
+        :param lcs: Local coordinate system
+        :return: ---
+        """
+        self._graph.add_edge(node_from, node_to, lcs=lcs)
+        self._graph.add_edge(node_to, node_from, lcs=lcs.invert())
+
+    def _check_coordinate_system_exists(self, coordinate_system_name: Hashable):
+        """
+        Raise an exception if the specified coordinate system does not exist.
+
+        :param coordinate_system_name: Name of the coordinate system, that should be
+        checked.
+        :return: ---
+        """
+        if not self.has_coordinate_system(coordinate_system_name):
+            raise Exception(
+                "There is no coordinate system with name " + str(coordinate_system_name)
+            )
```

Comment:
I think raising [`ValueError`](https://docs.python.org/3.7/library/exceptions.html#ValueError) (or `KeyError`?) is more fitting in these cases